### PR TITLE
Properly shutdown clients in reconnect tests

### PR DESCRIPTION
**Pull request by mdumandag**
Client shutdown calls were missing for some of the reconnect tests.
Added that and added a secondary shutdown mechanism in case something
goes wrong in the tests (like assertion failures) and we couldn't shutdown
the client.

### CODE REVIEW

File: tests/integration/backward_compatible/reconnect_test.py [hunk 1]
```diff
@@ -268,3 +280,6 @@ def assertion():
             self.assertNotEqual(0, event_count.get())
 
         self.assertTrueEventually(assertion)
+
+        client.shutdown()
+        self.rc.terminateCluster(cluster.id)
```
Comment (srknzl) [inline tests/integration/backward_compatible/reconnect_test.py:285]: how was the client shutting down before this added?

File: tests/integration/backward_compatible/reconnect_test.py [hunk 2]
```diff
@@ -244,6 +254,8 @@ def _verify_listeners_after_client_disconnected(self, member_address, client_add
             cluster_connect_timeout=sys.maxsize,
         )
 
+        self.client = client
+
```
Comment (srknzl) [inline tests/integration/backward_compatible/reconnect_test.py:258]: nit: could do this as well (same for other `self.client = client`):
```python
self.client = HazelcastClient(
            cluster_name=cluster.id,
            cluster_members=[client_address],
            cluster_connect_timeout=sys.maxsize,
 )
```

**Review by srknzl — APPROVED** [review]
LGTM 👍🏻  Left a question and a nit

File: tests/integration/backward_compatible/reconnect_test.py (hunk 1 above)
Comment (mdumandag) [reply tests/integration/backward_compatible/reconnect_test.py:285]: It was not :D That was the problem

Comment (srknzl) [reply tests/integration/backward_compatible/reconnect_test.py:285]: ok that question was to clarify 😄 , thanks

File: tests/integration/backward_compatible/reconnect_test.py (hunk 2 above)
Comment (mdumandag) [reply tests/integration/backward_compatible/reconnect_test.py:258]: Then I head to reference them as self.client instead of client, which I don't like

**Comment by mdumandag** [comment]
Thanks for the review Serkan

**Pull request merged**